retail/interactive-tutorial/RetailProducts.Samples/UpdateProductSample.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Copyright 2021 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START retail_update_product]
+// Update product from a catalog using Retail API
+
+using Google.Cloud.Retail.V2;
+using System;
+using System.Linq;
+
+/// <summary>
+/// The update product sample class.
+/// </summary>
+public class UpdateProductSample
+{
+    private static readonly Random Random = new Random();
+    private static readonly string GeneratedProductId = RandomAlphanumericString(14);
+
+    /// <summary>
+    /// Generate the random alphanumeric string.
+    /// </summary>
+    /// <param name="length">The required length of alphanumeric string.</param>
+    /// <returns>Generated alphanumeric string.</returns>
+    private static string RandomAlphanumericString(int length)
+    {
+        const string chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+        return new string(Enumerable.Repeat(chars, length)
+            .Select(s => s[Random.Next(s.Length)]).ToArray());
+    }
+
+    /// <summary>
+    /// Generate product for update.
+    /// </summary>
+    /// <param name="productId">The actual product id.</param>
+    /// <param name="projectId">The current project id.</param>
+    /// <returns>Generated retail product.</returns>
+    private static Product GenerateProductForUpdate(string productId, string projectId)
+    {
+        var updatedPriceInfo = new PriceInfo
+        {
+            Price = 20.0f,
+            OriginalPrice = 25.5f,
+            CurrencyCode = "EUR"
+        };
+
+        var generatedProduct = new Product
+        {
+            Id = productId,
+            Name = $"projects/{projectId}/locations/global/catalogs/default_catalog/branches/default_branch/products/{productId}",
```

Comment:
In general, we try that users don't need to know how resource names are formed.
```suggestion
            ProductName = new ProductName(projectId, "global", "default_catalog", "default_branch", productId),
```

Reply:
Removed logic which generates another product object inside **Update** sample, so we no longer have this issue with the ProductName.